src/main/java/seedu/address/logic/commands/ScheduleCommand.java

```diff
@@ -0,0 +1,171 @@
+        this.targetIndex = targetIndex;
+        this.doBefore = doBefore;
+        this.expectedTime = expectedTime;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Assignment> lastShownList = model.getFilteredAssignmentList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_ASSIGNMENT_DISPLAYED_INDEX);
+        }
+
+        Assignment assignmentToSchedule = lastShownList.get(targetIndex.getZeroBased());
+
+        Schedule schedule = createValidSchedule(assignmentToSchedule, model.getFilteredTaskList());
+
+        Assignment scheduledAssignment = createScheduledAssignment(assignmentToSchedule, schedule);
+
+        model.setAssignment(assignmentToSchedule, scheduledAssignment);
+        return new CommandResult(String.format(MESSAGE_SCHEDULE_ASSIGNMENT_SUCCESS, scheduledAssignment));
+    }
+
+    private Schedule createValidSchedule(Assignment assignmentToSchedule, List<Task> taskList) throws CommandException {
+        LocalDateTime newDeadline = doBefore.toLocalDateTime();
+        if (assignmentToSchedule.getDeadline().isBefore(doBefore)) {
+            newDeadline = assignmentToSchedule.getDeadline().toLocalDateTime();
+        }
+
+        LocalDateTime nearestTime = roundToHour(LocalDateTime.now().plusHours(1));
+
+        List<LocalDateTime> possibleTime = generateAllPossibleTime(nearestTime,
+                newDeadline, taskList);
+
+        if (possibleTime.isEmpty()) {
+            throw new CommandException("No possible schedule");
+        }
+        return getRandom(possibleTime);
+    }
+
+    private LocalDateTime roundToHour(LocalDateTime time) {
+        int min = time.getMinute();
+        time = time.minusMinutes(min % 60);
+        return time;
+    }
+
+    private Schedule getRandom(List<LocalDateTime> list) {
+        int rnd = new Random().nextInt(list.size());
+        Deadline suggestedStartTime = new Deadline(list.get(rnd));
+        Deadline suggestedEndTime = new Deadline(list.get(rnd).plusHours(expectedTime));
+        return new Schedule(suggestedStartTime, suggestedEndTime);
+    }
+
+    private List<LocalDateTime> generateAllPossibleTime(LocalDateTime start, LocalDateTime end, List<Task> taskList) {
+        List<LocalDateTime> possibleTime = new ArrayList<>();
+        for (LocalDateTime i = start; i.isBefore(end); i = i.plusHours(1)) {
+            boolean can = true;
+            // working hours
+            if (!isWorkingHour(i, i.plusHours(expectedTime))) {
+                can = false;
+            }
+            // no overlap
+            for (Task j: taskList) {
+                if (overlap(i, i.plusHours(expectedTime), j)) {
+                    can = false;
+                }
+            }
+
+            if (can) {
+                possibleTime.add(i);
+            }
+        }
+        return possibleTime;
+    }
```

Comment:
Good job :) A suggestion you may want to consider is to name the boolean variable "can" to something more descriptive and meaningful, as currently "can" doesn't really describe what the boolean is checking for. If you do change the name of the variable, remember to adhere to the CS2103T conventions and ensure that it sounds like a boolean value (i.e isPossibleTime, canSchedule etc)

Reply:
Noted! Thanks